vars/mavenExecute.groovy

```diff
@@ -1,139 +1,25 @@
-    'globalSettingsFile',
-    /** Path or url to the mvn settings file that should be used as project settings file.*/
-    'projectSettingsFile',
-    /** Path to the pom file that should be used.*/
-    'pomPath',
-    /** Path to the location of the local repository that should be used.*/
-    'm2Path'
-]
-@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS.plus([
-    /** @see dockerExecute */
-    'dockerOptions',
-    /** Flags to provide when running mvn.*/
-    'flags',
-    /** Maven goals that should be executed.*/
-    'goals',
-    /** Additional properties.*/
-    'defines',
-    /**
-     * Configures maven to log successful downloads. This is set to `false` by default to reduce the noise in build logs.
-     * @possibleValues `true`, `false`
-     */
-    'logSuccessfulMavenTransfers',
-    /**
-     * Returns the standard output of the maven command for further processing. By default stdout will be printed to the console but not returned by mavenExecute.
-     * @possibleValues `true`, `false`
-     */
-    'returnStdout'
-])
-
-/**
- * Executes a maven command inside a Docker container.
- */
-@GenerateDocumentation
 def call(Map parameters = [:]) {
-
     final script = checkScript(this, parameters) ?: this
-    String commandOutput = ''
-
-    // load default & individual configuration
-    Map configuration = ConfigurationHelper.newInstance(this)
-        .loadStepDefaults()
-        .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
-        .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
-        .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName ?: env.STAGE_NAME, STEP_CONFIG_KEYS)
-        .mixin(parameters, PARAMETER_KEYS)
-        .use()
+    parameters = DownloadCacheUtils.injectDownloadCacheInMavenParameters(script, parameters)
 
-    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+    List credentials = [ ]
+    piperExecuteBin(parameters, STEP_NAME, METADATA_FILE, credentials)
 
-        new Utils().pushToSWA([
-            step         : STEP_NAME,
-            stepParamKey1: 'scriptMissing',
-            stepParam1   : parameters?.script == null
-        ], configuration)
-
-        String command = "mvn"
-
-        String globalSettingsFile = configuration.globalSettingsFile?.trim()
-        if (globalSettingsFile) {
-            if (globalSettingsFile.startsWith("http")) {
-                globalSettingsFile = downloadSettingsFromUrl(this, globalSettingsFile, 'global-settings.xml')
-            }
-            command += " --global-settings '${globalSettingsFile}'"
-        }
-
-        String m2Path = configuration.m2Path
-        if (m2Path?.trim()) {
-            command += " -Dmaven.repo.local='${m2Path}'"
+    String output = ''
+    if (parameters.returnStdout) {
+        String outputFile = '.pipeline/maven_output.txt'
+        if (!fileExists(outputFile)) {
```

Comment:
As said above. If output is empty the output file will not exist and thus fail here with a error message users might not understand.